### PR TITLE
8288619: Unexpected parsing for @see

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

When invoking the method `DocCommentParser::quotedString` with field `newline` as `true` 
and the quoted string has the character `@`, `quotedString` will break the loop and return `null`.
This patch changes the field `newline` to `false` at the beginning of the method `quotedString`
to solve this issue. And a corresponding test case is added.

Thanks for the review.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288619](https://bugs.openjdk.org/browse/JDK-8288619): Unexpected parsing for @see

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13888/head:pull/13888` \
`$ git checkout pull/13888`

Update a local copy of the PR: \
`$ git checkout pull/13888` \
`$ git pull https://git.openjdk.org/jdk.git pull/13888/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13888`

View PR using the GUI difftool: \
`$ git pr show -t 13888`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13888.diff">https://git.openjdk.org/jdk/pull/13888.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13888#issuecomment-1540206811)

### CODE REVIEW

**Comment by jonathan-gibbons** [comment]
> > Overall, I would expect `newline` to be set `false`earlier, maybe in `blockTag`, so that the setting applies to all block tags. That might still leave the possibility of a newline happening after the tag name and before the quoted string, but your test case does not cover that.
> 
> If there is something between the tag name and the quoted string, the method `quotedString` is never invoked by method `TagParser::parse`. So such case seems not related to this bug. Do I miss anything?

I was just trying to imagine a case where `newline` might be true after the tag name has been read.

**Review by jonathan-gibbons — CHANGES_REQUESTED** [review]
I see the problem reported in the JBS issue, but the proposed fix is not the best one.

It would be better to set the `newline` flag to false earlier, perhaps at the beginning of `blockTag`

File: test/langtools/tools/javac/doctree/SeeTest.java [hunk 1]
```diff
@@ -50,6 +50,24 @@ void quoted_text() { }
         Text[TEXT, pos:12, "String"]
     ]
 ]
+*/
+
+    /**
+     * Test '@' in quoted string.
+     * @see "{@code}"
+     */
+    void at_sign_in_quote_string() { }
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/doctree/SeeTest.java:59]: (minor) add `d` after `quote` in method name

Comment (jonathan-gibbons) [reply test/langtools/tools/javac/doctree/SeeTest.java:59]: Also, add a test case for this, with a newline after the tag:
```

    /**
     * abc.
     * @see
     *    "{@code ...}"
     */
```

Comment (lgxbslgx) [reply test/langtools/tools/javac/doctree/SeeTest.java]: > (minor) add `d` after `quote` in method name

Fixed.

**Comment by lgxbslgx** [comment]
> It would be better to set the `newline` flag to false earlier, perhaps at the beginning of `blockTag`

Fixed.

> Here is another test case:
> 
> ```
>     /**
>      * case 2.
>      * @see
>      *    "{@code ...}"
>      */
>     public void m2() { }
> ```

I add this test case. But it fails with the following error message. The `DocCommentTester#PrettyChecker` seems not work as expected?

```
METHOD new_line_before_quoted_string PrettyChecker
Error: mismatch
*** expected:
Test_new_line_before_quoted_string.
@see
____"{@code}"
*** found:
Test_new_line_before_quoted_string.
@see_"{@code}"
```

**Review by jonathan-gibbons — APPROVED** [review]

**Comment by lgxbslgx** [comment]
@jonathan-gibbons Thanks for the review. Integrating.

/integrate

**Pull request closed without merging**